python/img_hdf5_reader.py

```diff
@@ -187,6 +195,74 @@ def to_ometiff(self, channel_clips, sample, json_file):
 
         io.write_ometiff(ometif_path, image, str(omexml))
 
+    def to_zarr(self, channel_clips, sample, json_file):
+        # zarr default BLOSC not supported in browser
+        COMPRESSOR = Zlib(level=1)
+        MAX_ALLOWED = 65536
+        TILE_SIZE = 512
+
+        channels = [channel for (channel, clip) in channel_clips]
+        data_shape, data_dtype = self._get_shape_and_dtype(channels)
+
+        # Create mutable store
+        zarr_path = f"{json_file.name.replace('.json', '')}.zarr"
+        pyramid_group = "pyramid"
+        store = zarr.DirectoryStore(zarr_path)
+        root = zarr.group(store=store, overwrite=True)
+        root.create_group(pyramid_group, overwrite=True)
+
+        channel_data = {}
+        images = []
+        s3_target = open("s3_target.txt").read().strip()
+        for idx, (channel, clip) in enumerate(channel_clips):
+            channel_data[channel] = {
+                "sample": sample,
+                "tileSource": "https://s3.amazonaws.com/{}/linnarsson/".format(
+                    s3_target
+                )
+                + "linnarsson.images.zarr/pyramid/",
+            }
+
+            channels.append(channel)
+            array = self.scale_sample(
+                channel=channel,
+                sample=sample,
+                max_allowed=MAX_ALLOWED,
+                clip=float(clip),
+                use_dask=True,
+            ).astype(data_dtype)
+
+            images.append(array.T)
+
+        # Stack arrays using dask and rechunk to tile sizes (i.e. (2, 512, 512)
+        chunks = (len(channels), TILE_SIZE, TILE_SIZE)
+        da.array(images).rechunk(chunks).to_zarr(
+            zarr_path, component=f"{pyramid_group}/00", compressor=COMPRESSOR
+        )
+        json.dump(channel_data, json_file, indent=2)
+
+    def _get_shape_and_dtype(self, channels):
+        shapes, dtypes = zip(
+            *[(self.data[c].shape, self.data[c].dtype) for c in channels]
+        )
+        # Compare dimesion sizes to make sure all same size
+        shape = []
+        for shared_dimension in zip(*shapes):
+            first_el = shared_dimension[0]
+            all_same = 1
+            if not all_same:
+                raise ValueError(
+                    f"Data stored in {channels} are not the same shape."
+                )
+            shape.append(first_el)
+        # Ensure same dtype.
+        first_dtype = dtypes[0]
+        all_same = all(dtype == first_dtype for dtype in dtypes)
+        if not all_same:
+            raise ValueError(f"Dtypes are not the same for {channels}.")
```

Comment:
This is clean, but would it be better to iterate over the dtypes, so if there was a problem, we could say exactly where it was? Though if that's unlikely, not worth the change.

Reply:
I think this is unlikely

Reply:
I changed this to print both the `channels` and `dtypes` so it's easier to debug.